internal/corazawaf/body_buffer.go

```diff
@@ -99,30 +100,45 @@ type bodyBufferReader struct {
 }
 
 func (b *bodyBufferReader) Read(p []byte) (n int, err error) {
+	if b.br == nil {
+		// reader has been closed and hence we don't attempt to do anymore read
+		return 0, io.EOF
+	}
+
 	if !environment.HasAccessToFS || b.br.writer == nil {
 		buf := b.br.buffer.Bytes()
+
 		n = len(p)
-		if b.pos+n > len(buf) {
-			n = len(buf) - b.pos
+		if bl := b.br.buffer.Len(); b.pos+n > bl {
```

Comment:
Is this change needed? `len(buf)` seemed more obvious since `buf` is what we're reading from

Reply:
Not needed. It was part of my attempts to understand the logic.

Reply:
reverted